src/test/ui/error-codes/E0283.stderr

```diff
@@ -1,10 +1,16 @@
-error[E0283]: type annotations needed
+error[E0790]: cannot call associated function on trait without specifying the corresponding `impl` type
   --> $DIR/E0283.rs:30:21
    |
+LL |     fn create() -> u32;
+   |     ------------------- `Generator::create` defined here
+...
 LL |     let cont: u32 = Generator::create();
-   |                     ^^^^^^^^^^^^^^^^^ cannot infer type
+   |                     ^^^^^^^^^^^^^^^^^ cannot call associated function of trait
+   |
+help: use a fully-qualified path to a specific available implementation (2 found)
    |
-   = note: cannot satisfy `_: Generator`
+LL |     let cont: u32 = <::Impl as Generator>::create();
```

Comment:
This `::Impl` path is kinda funky, is there a reason we used `to_string_no_crate_verbose` instead of just `tcx.def_path_str`?

Reply:
I don't recall exactly anymore, but if I remember correctly I tried that and got paths as a result that didn't make sense in the context where they were suggested. But I could have made a mistake then. If you'd like I could try it again when I find the time and make a follow-up PR with that or at least find out why it didn't work in this case.

Reply:
Yeah, either put up a follow up or I can also look into it.

Reply:
(I was involved in the development of this PR)
I think it was along the lines of it returning `Impl` instead of the context-sensitive value `inner::Impl` in the case of nested modules.